### ISSUE TITLE
Choosing a testing library for AbSolute

### ISSUE THREAD

**Issue by ptal** [mpelleau/AbSolute#6]
AbSolute does not seem to have a test library/framework yet, and I would need one to make simple tests on the octagon abstract domain I'm developing.
Since I won't (shouldn't ? ;-)) be the only one to use it, I prefer to ask what is your preference.
After a quick look on the ones available in OPAM, my preference goes to [ppx_inline_test](https://github.com/janestreet/ppx_inline_test) for the following reasons:

* Very easy to write tests (maybe the most important reason, otherwise we don't write them...).
* Use OCaml syntax.
* Tests can be written inside the source file, which has various advantages such as:
  * Serving as valuable documentation.
  * Easy access to private information of modules (for example to check invariants of internal data structures).
* Developed and used by Jane Street. It is one of the most downloaded test framework and seem to be regularly updated.

What do you think?

**Comment by MarechalAlex** [comment]
There is a "make check" command to run some non-regression tests. If think they are handmade, meaning that they don't use any existing testing framework.

What kind of tests do you plan to make? Unit tests of functions?

**Comment by ptal** [comment]
Yes, unit tests. Black box tests are good and necessary too, but not for details / invariants of algorithms / ...

**Comment by ptal** [comment]
I use [alcotest](https://github.com/mirage/alcotest/) on my fork, which is not invasive, without ppx macro, and seems to work great so far!

**Comment by mpelleau** [comment]
Do you want to deploy it?

**Comment by ptal** [comment]
I forgot to push the changes on my fork but I already set up everything [here](https://github.com/ptal/AbSolute) (the build file, and the test/ repository), so we just need to pull-request this part (although I don't know how to pull request only that part and not the rest...).
